src/Controller/Api/Legacy/GroupController.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Controller\Api\Legacy;
+
+@trigger_error(
+    'The '.__NAMESPACE__.'\GroupController class is deprecated since sonata-project/user-bundle 4.x and will be removed in version 5.0.'
```

Comment:
```suggestion
    'The '.__NAMESPACE__.'\GroupController class is deprecated since sonata-project/user-bundle 4.x and will be removed in 5.0.'
```
It is nesessery? Mybe we should add templates for deprecations to keep some standards? @phansys WDYT? You are the best in it.

Reply:
>It is nesessery?

I'm afraid yes, since is the only way I know to tell users that this class will be not available in the next major release.

>Mybe we should add templates for deprecations to keep some standards? @phansys WDYT?

I think we could use [symfony/deprecation-contracts](https://github.com/symfony/deprecation-contracts) as starting point.

>You are the best in it.

I'm sorry to disagree, but surely I'm not the best. Regarding deprecations, exceptions, etc, I'm only putting the focus in leveraging the message structures and the wording already established by the language and other packages in the ecosystem. Trying to be as semantic, explicit and clear as possible.